kotlinpoet/src/jvmMain/kotlin/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -142,23 +142,24 @@ public class TypeSpec private constructor(
         codeWriter.emitCode("object")
         val supertype = if (superclass != ANY) {
           if (!areNestedExternal && !modifiers.contains(EXPECT)) {
-            listOf(CodeBlock.of(" %T(%L)", superclass, superclassConstructorParametersBlock))
+            listOf(CodeBlock.of("%T(%L)", superclass, superclassConstructorParametersBlock))
           } else {
-            listOf(CodeBlock.of(" %T", superclass))
+            listOf(CodeBlock.of("%T", superclass))
           }
         } else {
           listOf()
         }
 
-        val allSuperTypes = supertype + if (superinterfaces.isNotEmpty()) {
-          superinterfaces.keys.map { CodeBlock.of(" %T", it) }
-        } else {
-          emptyList()
+        val allSuperTypes = supertype + superinterfaces.entries.map { (type, init) ->
+          if (init == null) {
+            CodeBlock.of("%T", type)
+          } else {
+            CodeBlock.of("%T by %L", type, init)
+          }
         }
 
         if (allSuperTypes.isNotEmpty()) {
-          codeWriter.emitCode(" :")
-          codeWriter.emitCode(allSuperTypes.joinToCode(","))
+          codeWriter.emitCode(allSuperTypes.joinToCode(separator = ", ", prefix = " : "))
```

Comment:
Nice use of `joinToCode()`!

nit - this is the default separator so isn't necessary:

```suggestion
          codeWriter.emitCode(allSuperTypes.joinToCode(prefix = " : "))
```

Reply:
My personal philosophy would be to keep the separator as-is. We "want" it to be `", "`, the fact that it is also the default behavior is incidental.

Reply:
Actually, I was thinking about this like using a library, but this is actually defined in this project, so I'll change it. Thanks for the review

Reply:
Agree, specifying the argument explicitly doesn't hurt either, but yeah, mostly from the perspective that we own that method, plus `", "` seems to be a pretty standard separator in "joinTo" Kotlin methods.

Reply:
And like Kotlin's `joinToString`, we can never really change the default value or it would basically break all of its users.